File: 2-Regression/Simple Linear Regression/simple_linear_regression.R
# Simple Linear Regression

# Importing the dataset
dataset = read.csv('Salary_Data.csv')

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary , SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)


#performing a simple Linear Regression model
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)
# you can type summary(regressor) on console to check the P value it should be less than 5% to show enough relationship between traing set data (we found a valid line )

# predicting the test value
y_pred = predict(regressor, newdata = test_set)

#visualize the results for the training set
install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience , y =training_set$Salary),
             colour = 'red') +
  geom_line(aes(x= training_set$YearsExperience , y= predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('salary vs. experience training set') +
  xlab('years of experience') +
  ylab('salary')

ggplot() +
  geom_point(aes(x = test_set$YearsExperience , y =test_set$Salary),
             colour = 'black') +
  geom_line(aes(x= training_set$YearsExperience , y= predict(regressor, newdata = training_set)),
            colour = 'red') +
  ggtitle('salary vs. experience testing set') +
  xlab('years of experience') +
  ylab('salary')

File: 2-Regression/Simple Linear Regression/simple_linear_regression.py
# Simple Linear Regression


# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

#Fitting simple Linear Regression to training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

#predicting the Test set result
y_pred = regressor.predict(X_test)

#visualise training set result
plt.scatter(X_train,y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs. Experince (Training set)')
plt.xlabel('years of experience')
plt.ylabel('Salary')
plt.show()

#visualise Testing set result
plt.scatter(X_test,y_test, color = 'black')
plt.plot(X_train, regressor.predict(X_train), color = 'red')
plt.title('Salary vs. Experince (Testing set)')
plt.xlabel('years of experience')
plt.ylabel('Salary')
plt.show()
